### PR TITLE
Respect noindex meta tags and do not include them in the sitemap

### PR DESCRIPTION
**Pull request by jobackman**
Hi, 
First I'd like to say thanks for a great tool! 

From what I understand about SEO though I believe it's preferable to exclude pages with 
```html
<meta name="robots" content="noindex, follow">
```
from being included in a websites sitemap. So I've made an attempt at making the generator respect robots meta tags that have `noindex` values. This should allow pages to be crawled but not included in the sitemap if it has `noindex` in its robots tag.

TLDR on what it does: 

 - Defines `crawler.noindex` as an empty array meant to contain all the links crawled that should be excluded from the sitemap. 
 - Identifies whether it is a `noindex`-page during the crawler's `discoverResources`
 - Before building the XML removes the `noindex` urls.

From what I have tried it seems to be working as it should. It should also resolve https://github.com/lgraubner/sitemap-generator-cli/issues/8 if it's published.

### CODE REVIEW

**Comment by lgraubner** [comment]
Thanks, looks great!

I added some lines to add the removed URL to the ignored array and provided a test case.
